apis/storage/v1alpha1/anywherecache_types.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var StorageAnywhereCacheGVK = GroupVersion.WithKind("StorageAnywhereCache")
+
+// StorageAnywhereCacheSpec defines the desired state of StorageAnywhereCache
+// +kcc:spec:proto=google.storage.control.v2.AnywhereCache
+type StorageAnywhereCacheSpec struct {
+	// Immutable. The reference to bucket where cache needs to be created.
+	// +required
+	BucketRef *refs.StorageBucketRef `json:"bucketRef"`
+
+	// Immutable. The zone in which the cache instance needs to be created. For example, us-central1-a.
+	// +required
+	Zone *string `json:"zone"`
+
+	// The AnywhereCacheID generated via backend, It can be used by users to manage an existing cache.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// The desired state of the cache. Possible values include "running", "disabled", and "paused".
+	// If not specified, the default value is "running". This field controls the runtime behavior of the cache.
+	// Please note that changes to the `desiredState` are prioritized over any other updates.
+	// For instance, if both the `desiredState` and `ttl` are updated simultaneously, the state would be
+	// updated first, followed by `ttl`.
+	DesiredState *string `json:"desiredState,omitempty"`
+
+	// Cache entry TTL (ranges between 1h to 7d). This is a cache-level config
+	//  that defines how long a cache entry can live. Defaults to "86400s"
+	//  TTL must be in whole seconds.
+	Ttl *string `json:"ttl,omitempty"`
+
+	// Cache admission policy. Valid values includes:
+	//  `admit-on-first-miss` and `admit-on-second-miss`. Defaults to
+	//  `admit-on-first-miss`.
+	AdmissionPolicy *string `json:"admissionPolicy,omitempty"`
```

Comment:
these 3 fields all have defaults so they will never be "unset" effectively. Let's define default values via kubebuilder tags https://book.kubebuilder.io/reference/markers/crd-validation